components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/jdbc/JDBCUserStoreCountRetriever.java

```diff
@@ -178,7 +181,23 @@ public Long countClaim(String claimURI, String valueFilter) throws UserStoreCoun
     private Connection getDBConnection(RealmConfiguration realmConfiguration) throws SQLException, UserStoreException {
 
         Connection dbConnection = null;
-        DataSource dataSource = DatabaseUtil.createUserStoreDataSource(realmConfiguration);
+        String dataSourceName = realmConfiguration.getUserStoreProperty(JDBCRealmConstants.DATASOURCE);
+        if (dataSourceName == null) {
+            dataSourceName = realmConfiguration.getUserStoreProperties().get("DomainName");
```

Comment:
when a secondary userstore is configured the Datasource name(jndi name) will be null, hence we thought of using domainName as the datasource name in case of a secondary user store

Reply:
This issue was basically reported because of this, as the datasource name is null, it creates new datasource each time. To avoid this we used the domain name as the datasource name.

Reply:
This does not explain how "dataSourceName" is equal to the value configured for "DomainName"

Reply:
This datasourceName variable is used to store datasource in the property Map. Hence in case when datasource name is null userstore name is considered as datasource name to uniquely identify the datasource. So I think its fine to use the same variable name.

Reply:
Add a comment explaining in what scenarios this can happen.